internal/v5/api.go

```diff
@@ -1120,21 +1120,65 @@ func (h *ReqHandler) serveMacaroon(_ http.Header, _ *http.Request) (interface{},
 // GET /delegatable-macaroon
 // See https://github.com/juju/charmstore/blob/v4/docs/API.md#get-delegatable-macaroon
 func (h *ReqHandler) serveDelegatableMacaroon(_ http.Header, req *http.Request) (interface{}, error) {
+	values, err := url.ParseQuery(req.URL.RawQuery)
+	if err != nil {
+		return nil, errgo.Mask(err)
+	}
+	entityIds := values["id"]
+	if len(entityIds) == 0 {
+		auth, err := h.authorize(req, []string{params.Everyone}, true, nil)
```

Comment:
// No entity ids, so we provide a macaroon that's good for any entity that the
// user can access, as long as that entity doesn't have terms and conditions.

?